src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -54,6 +55,13 @@ private TypeElement getElement(Class<?> clazz) {
     return compilation.getElements().getTypeElement(clazz.getCanonicalName());
   }
 
+  private boolean isJava8() {
+    String version = System.getProperty("java.version");
+    int pos = version.indexOf('.');
+    pos = version.indexOf('.', pos+1);
+    return Double.valueOf(version.substring (0, pos)).compareTo(Double.valueOf(1.8)) >= 0;
```

Comment:
The way I prefer to do this sort of thing is lame, it's just:

```
try {
  Modifier.valueOf("DEFAULT")
  return true;
} catch (Exception e) {
  return false;
}
```
That way you're testing for the feature you want, not the version it arrived in. This also works very nicely for Android.